src/main/java/com/codereview/ipcounter/service/impl/UniqueIpCounter.java

```diff
@@ -0,0 +1,45 @@
+package com.codereview.ipcounter.service.impl;
+
+import com.codereview.ipcounter.service.IpCounter;
+import com.codereview.ipcounter.utils.ParserUtils;
+import java.net.UnknownHostException;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+@Slf4j
+@Service
+public class UniqueIpCounter implements IpCounter<String> {
+
+  private final int[] seen = new int[1 << 27];
+
+  private long count;
+
+  @Override
+  public void countIp(String ipView) {
+    try {
+      var longIpView = ParserUtils.toLongValue(ipView);
+      registerIfNewUnique(longIpView);
+    } catch (UnknownHostException e) {
+      log.error("An error occurred during try to parse ip {}", ipView, e);
```

Comment:
Warning
>log cannot be resolved

Reply:
Awful. You skip this ip and go next. Is it right logic?